### PR TITLE
[for CI] make ExactSequence more readable

### PR DESCRIPTION
**Pull request by Treeston**

### CODE REVIEW

**Comment by Treeston** [comment]
with thanks to @Shauren

File: src/server/game/Chat/ChatCommands/ChatCommandTags.h [hunk 1]
```diff
@@ -45,6 +47,35 @@ namespace Trinity::Impl::ChatCommands
...
+        return i >= N ? '\0' : s[i];
+    }
+
+#define CHATCOMMANDS_IMPL_SPLIT_LITERAL_EXTRACT_CHAR(z, i, strliteral) \
+        BOOST_PP_COMMA_IF(i) Trinity::Impl::ChatCommands::GetChar(strliteral, i)
+
+#define CHATCOMMANDS_IMPL_SPLIT_LITERAL_CONSTRAINED(maxlen, strliteral)  \
+        BOOST_PP_REPEAT(maxlen, CHATCOMMANDS_IMPL_SPLIT_LITERAL_EXTRACT_CHAR, strliteral)
+
+    // this creates always 50 elements - "abc" -> 'a', 'b', 'c', '\0', '\0', ... up to 50
```
Comment (Shauren) [inline src/server/game/Chat/ChatCommands/ChatCommandTags.h:77]: hah, you didnt even bother to change my original comment

File: src/server/game/Chat/ChatCommands/ChatCommandTags.h [hunk 2]
```diff
@@ -45,6 +47,35 @@ namespace Trinity::Impl::ChatCommands
...
+        size_t i = 0;
+        for (; i < (sizeof...(chars)); ++i)
+            if (str[i] == '\0')
+                break;
+
+        return i;
+    }
+
+    template <unsigned int N>
+    inline constexpr char GetChar(char const (&s)[N], unsigned int i)
```
Comment (Shauren) [inline src/server/game/Chat/ChatCommands/ChatCommandTags.h:65]: this should have been size_t everywhere (N and i)

Comment (Shauren) [inline src/server/game/Chat/ChatCommands/ChatCommandTags.h:65]:
```suggestion
    template <size_t N>
    inline constexpr char GetChar(char const (&s)[N], size_t i)
```

File: src/server/game/Chat/ChatCommands/ChatCommandTags.h [hunk 3]
```diff
@@ -45,6 +47,35 @@ namespace Trinity::Impl::ChatCommands
...
+    {
+        static_assert(N <= 20, "SPLIT_LITERAL macro can only be used with up to 50 character long literals");
```
Comment (Shauren) [inline src/server/game/Chat/ChatCommands/ChatCommandTags.h:67]:
```suggestion
        static_assert(N <= 20, "SPLIT_LITERAL macro can only be used with up to 20 character long literals");
```

**Pull request merged**